### PR TITLE
Add support for properties required for HierarchyTable Depth scenarios.

### PR DESCRIPTION
**Pull request by jkerihuel**

### CODE REVIEW

File: mapiproxy/servers/default/emsmdb/emsmdbp_object.c [hunk 1]
```diff
@@ -2735,17 +2739,43 @@ static int emsmdbp_object_get_properties_mapistore_root(TALLOC_CTX *mem_ctx, str
...
 
 	contextID = emsmdbp_get_contextID(object);
 
 	folder = (struct emsmdbp_object_folder *) object->object.folder;
         for (i = 0; i < properties->cValues; i++) {
-                if (properties->aulPropTag[i] == PR_CONTENT_COUNT) {
+		if (properties->aulPropTag[i] == PidTagFolderFlags) {
+			/* Folder flags / yet undocumented */
+			folder_flags = talloc_zero(data_pointers, uint32_t);
+			*folder_flags = 0x5;
```
Comment (sixstone-qq) [inline mapiproxy/servers/default/emsmdb/emsmdbp_object.c]: I would be great to have this as #define something and document it :)

Comment (jkerihuel) [inline mapiproxy/servers/default/emsmdb/emsmdbp_object.c]: I have specifically added "yet undocumented" for this reason. The property is not documented in Microsoft specifications and I am waiting for more information about it for Microsoft.

File: mapiproxy/servers/default/emsmdb/emsmdbp_object.c [hunk 2]
```diff
@@ -1898,6 +1899,9 @@ _PUBLIC_ void **emsmdbp_object_table_get_row_props(TALLOC_CTX *mem_ctx, struct e
 				case PidTagDeletedCountTotal:
 				case PidTagAccess:
 				case PidTagAccessLevel:
+				case PidTagFolderFlags:
+				case 0x40820040:
```
Comment (sixstone-qq) [inline mapiproxy/servers/default/emsmdb/emsmdbp_object.c]: Could we add this constant somewhere in a .h file? Is it a named prop?

Comment (jkerihuel) [inline mapiproxy/servers/default/emsmdb/emsmdbp_object.c]: No we can't, yet undocumented and pending feedback from Microsoft.

**Comment by sixstone-qq** [comment]
It would be great to wait for a :+1: before merging unless too much rush is required.

**Comment by jkerihuel** [comment]
Sorry but I can't wait for a :+1: for this kind of pull requests. I have way too much pending code to push in my pipeline and I prefer PR review to be done on significant ones.

**Comment by blaxter** [comment]
For future commits, it really helps to add prefixes (like emsmdb_object or mapiproxy) to commit messages and also first line not to big would be nice

**Pull request merged**